File: C++/18870.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int main(void)
{
	int arr[1000000] = {0, };
	int sortArr[1000000] = {0, };
	int n;
	scanf("%d", &n);

	for(int i = 0; i < n; i++){
		scanf("%d", &arr[i]);
		sortArr[i] = arr[i];
	}

	sort(sortArr, sortArr + n);

	for(int i = 0; i < n; i++){

	}







	return 0;
}

File: 파이썬/18870.py
n = int(input())
arr = list(map(int, input().split()))

tmp = set(arr)
sortArr = list(tmp)
sortArr.sort()

a = {}
for i in range(len(sortArr)):
    a[sortArr[i]] = i

answer = ""

for i in range(n):
    answer += str(a[arr[i]]) + " "
    
print(answer)
